File: menuRenewal/메뉴리뉴얼/menuRenewal.js
/**공통문자열을 찾는 문제
 *  문자열의 요소로 조합 (combination) 을 만드는 방법: 모든 경우의 수를 따져보는 것이기 때문에 완전탐색이다.
 * 각각의 문자열을 순회하면서 공통문자열의 길이가 2이면, 새로운 배열에 공통문자열을 추가한다.
 *  중복을 배제하고, 입출력 예의 법칙을 따라 각각으 공통문자열은 사전순으로 정렬하여 추가
 */

function comb(arr, n) {
  let result = [];

  if (n === 1) return arr.map((value) => [value]);
  arr.forEach((p, idx, org) => {
    let leftover = org.slice(idx + 1);
    let combination = comb(rest, n - 1);
  });
}

function solution(orders, course) {
  let answer = [];
  return answer;
}

console.log(
  solution(['ABCFG', 'AC', 'CDE', 'ACDE', 'BCFG', 'ACDEH'], [2, 3, 4])
);

File: menuRenewal/메뉴리뉴얼/menurenewal.py
from itertools import combinations

def solution (orders, course):
  answer = []
  #코스요리 단품개수를 순회
  for dish in course:
    dict = {}

    # 
    for i in range(len(orders)):
      comb = [ ''.join(c) for c in combinations(sorted(orders[i]), dish)]
      for j in range(len(comb)):
        dict[comb[j]] = dict.get(comb[j], 0) +1

  # 딕셔너리를 순회하며 dict 안의 최대값을 구하고
  for k in dict:
    DICT_MAX = max(dict.values())
    #딕셔너리의 최대값이 2 이상이고 순회하는 각각 값이 최대값과 같을 경우
    if DICT_MAX >= 2 and dict[k] == DICT_MAX:
      answer.append(k)
  answer.sort()
  return answer

print(solution(["ABCFG", "AC", "CDE", "ACDE", "BCFG", "ACDEH"], [2,3,4]))